src/django/api/views/boundary.py

```diff
@@ -83,26 +83,46 @@ class BoundaryShapeView(APIView):
     permission_classes = [IsAuthenticated]
 
     def put(self, request, id, format=None):
+        boundary = BoundaryShapeView.get_boundary(request, id)
+        BoundaryShapeView.check_user_has_access_to_shape(request.user)
+        BoundaryShapeView.check_boundary_is_editable(boundary)
+
+        serializer = ShapeSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        boundary.latest_submission.shape = serializer.validated_data
+        boundary.latest_submission.save()
+
+        return Response(status=HTTP_204_NO_CONTENT)
+
+    def delete(self, request, id, format=None):
+        boundary = BoundaryShapeView.get_boundary(request, id)
+        BoundaryShapeView.check_user_has_access_to_shape(request.user)
+        BoundaryShapeView.check_boundary_is_editable(boundary)
+
+        boundary.latest_submission.shape = None
```

Comment:
This fails because currently the shape is not nullable:

https://github.com/azavea/iow-boundary-tool/blob/38ca58c0685cb7c0988f68f7ec35258e369d0bb1/src/django/api/models/submission.py#L19

We can either make that field nullable, or we can make it so that deletion only happens in the front-end, and we write a new shape the next time the user creates / uploads one. I have a preference for the second approach, because ensuring that every submission has a shape simplifies things downstream.

Reply:
Ah I see we're making that field nullable in #145. Once this is rebased on that it should start working correctly.

Reply:
I actually ran into this issue for different reasons in #145 and [made the field nullable there](https://github.com/azavea/iow-boundary-tool/pull/145/commits/9b089d8f8bf84bd3db29eac1af75106e9bb4b2c1#diff-c93adfd0016a9a82d8efb02214a793105a76ecc7c66f2f225ed8ddd73590f688). The migration is tied with other changes in a single commit. I suppose I could rebase and extract those changes into a separate commit if needed. 

I agree that ensuring every submission has a shape would be nice, but this could be a problem when creating a new boundary because we would need a default shape. And this shape would appear when a user first sees the draw page unless we add some logic to the draw page to check if this is a first-time-load, and if so, don't draw the polygon (or set it deleted in the front-end).

Reply:
You're right, I'd forgotten about creating a blank draft submission. Making it nullable is good 👍